Data base design

ars-ishi opened the pull request:
DB設計についてREADMEに記載をしました。確認をお願いいたします。

ars-ishi committed:
Write README

ars-ishi committed:
Edit README

yuzujoe commented on README.md:
```diff
@@ -1,24 +1,56 @@
-* Ruby version
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, unique: true|
```
user検索を後ほどするのでそのオプションが欲しいですね

ars-ishi replied:
users tableのnameに、index: trueを追加しました。

yuzujoe commented on README.md:
```diff
@@ -1,24 +1,56 @@
+#### Association
+- has_many :messages
+- has_many :groups, through: :group_users
+- has_many :group_users, dependent: :destroy
```
今回chat-spaceではグループを削除する昨日はなかったはずなのでこちらは不必要かと思います

ars-ishi replied:
users tableのアソシエーション
has_many :group_users, dependent: :destroyの
dependent: :destroyを削除しました。

yuzujoe commented on README.md:
```diff
@@ -1,24 +1,56 @@
-* Database initialization
+|Column|Type|Options|
+|------|----|-------|
+|chat|text|null: false|
```
chat-space必ずchat送らないとmessage送れない仕様でしたでしょうか？

ars-ishi replied:
messages tableのchat
null: falseを削除しました。

yuzujoe commented on README.md:
```diff
@@ -1,24 +1,56 @@
-* ...
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false, unique: true|
```
同じグループ名も今回は作成できますのでunique: trueは今回は不必要です。

ars-ishi replied:
groups tableのgroup_name
unique: trueを削除しました。

yuzujoe commented on README.md:
```diff
@@ -1,24 +1,56 @@
+#### Association
+- has_many :messages
+- has_many :groups, through: :group_users
+- has_many :group_users, dependent: :destroy
```
ここのdependent: :destroyもですね

ars-ishi replied:
groups tableのアソシエーション
has_many :group_users, dependent: :destroyから
dependent: :destroyを削除しました。

ars-ishi committed:
Edit README

変更点
・usersテーブルのアソシエーション
dependent: :destroy削除
・messagesテーブルのchat
null: false削除
・groupsテーブルのgroup_name
unique: true削除
・groupsテーブルのアソシエーション
dependent: :destroy削除

ars-ishi commented:
ご指摘いただいた点を修正いたしました。ご確認お願いいたします。

##修正点

1. users tableのname　/　index: trueを追加
2. users tableのアソシエーション　/　has_many :group_usersからdependent: :destroyを削除
3. messages tableのchat　/　null: falseを削除
4. groups tableのgroup_name　/　unique: trueを削除
5. groups tableのアソシエーション　/　has_many :group_usersからdependent: :destroyを削除

yuzujoe commented on README.md:
```diff
@@ -13,14 +13,14 @@
-|chat|text|null: false|
+|chat|text||
 |image|string||
 |group_id|integer|null: false, foreign_key: true|
```
integer型でもokですがreferences型の使い方が便利なのでそちらに直してしまいましょう

yuzujoe commented on README.md:
```diff
@@ -13,14 +13,14 @@
+|chat|text||
 |image|string||
 |group_id|integer|null: false, foreign_key: true|
 |user_id|integer|null: false, foreign_key: true|
```
こちらもですね

yuzujoe commented on README.md:
```diff
@@ -34,13 +34,13 @@
+- has_many :group_users
 
 
 ### group_users table
```
group_idとuser_idですね
